src/server/service/kv.rs

```diff
@@ -682,6 +722,47 @@ impl<T: RaftStoreRouter + 'static> tikvpb_grpc::Tikv for Service<T> {
         ctx.spawn(future);
     }
 
+    fn raw_batch_scan(
+        &self,
+        ctx: RpcContext,
+        mut req: RawBatchScanRequest,
+        sink: UnarySink<RawBatchScanResponse>,
+    ) {
+        const LABEL: &str = "raw_batch_scan";
+        let timer = self.metrics.raw_batch_scan.start_coarse_timer();
+
+        let (start_keys, end_keys): (Vec<_>, Vec<_>) = req.take_ranges()
+            .into_iter()
+            .map(|mut x| (x.take_start_key(), x.take_end_key()))
+            .unzip();
+
+        let future = self.storage
+            .async_raw_batch_scan(
```

Comment:
why not pass ranges directly?

Reply:
I'm not quite familiar with code base, so I just copy code skeleton from existing similar code. And so far, other places used separate start_key and end_key fields as well as arguments. Anyway, since it's ok to do so, let me change it this way.

Reply:
Updated